test/cmd/newapp.sh

```diff
@@ -165,16 +167,15 @@ os::cmd::expect_failure_and_text 'oc new-app -f test/testdata/template-with-name
 # ensure --build-env environment variables get added to the buildconfig
 os::cmd::expect_success_and_text 'oc new-app -f test/testdata/template-with-app-label.json --build-env FOO=bar -o yaml' 'FOO'
 # ensure the objects can actually get created with a namespace specified
-project=$(oc project -q)
 os::cmd::expect_success 'oc new-project template-substitute'
 os::cmd::expect_success 'oc new-project prefix-template-substitute'
-os::cmd::expect_success 'oc project ${project}'
+os::cmd::expect_success 'oc project ${default_project}'
 os::cmd::expect_success 'oc new-app -f test/testdata/template-with-namespaces.json -p SUBSTITUTED=template-substitute'
 os::cmd::expect_success 'oc delete all -l app=ruby-helloworld-sample'
 
 # ensure non-duplicate invalid label errors show up
-os::cmd::expect_failure_and_text 'oc new-app docker.io/library/wordpress -l qwer1345%$$#=self' 'error: ImageStream.image.openshift.io "wordpress" is invalid'
-os::cmd::expect_failure_and_text 'oc new-app docker.io/library/wordpress -l qwer1345%$$#=self' 'DeploymentConfig.apps.openshift.io "wordpress" is invalid'
+os::cmd::expect_failure_and_text 'oc new-app docker.io/library/wordpress -l qwer1345%$$#=self' 'error: ImageStream "wordpress" is invalid'
+os::cmd::expect_failure_and_text 'oc new-app docker.io/library/wordpress -l qwer1345%$$#=self' 'DeploymentConfig "wordpress" is invalid'
```

Comment:
why did you change this? it has nothing to do w/ your changes that i can see.

Reply:
I tried to use my locally built `oc` binary to run `cluster up`, but I cannot get this to work. I am running into various roadblocks when I do that. For example, I seem to be running into this issue - https://github.com/openshift/origin/issues/20263

I guess I would need some help on how to start a cluster using `oc` build from HEAD (including the use of locally build OpenShift images). I might not have the right workflow for this. 

Running the tests in the following setup:
```
$ oc version
oc v4.0.0-alpha.0+da45ed5-193-dirty
kubernetes v1.11.0+d4cacc0
features: Basic-Auth

Server https://192.168.99.109:8443
openshift v3.10.0+fd501dd-48
kubernetes v1.10.0+b81c8f8
```
I need to change the test assertions. Neither 'image.openshift.io' nor '.apps.openshift.io' appears in the output.

Not sure how to go about this. Are there any CI builds which run these tests? Just to verify that they do pass.

Reply:
it behaves as expected with a 3.11 server for me:

```
$ oc version
oc v4.0.0-alpha.0+c2e7c2a-184
kubernetes v1.11.0+d4cacc0
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://api.ci.openshift.org:443
openshift v3.11.0-alpha.0+010674d-1220
kubernetes v1.11.0+d4cacc0
```

```
--> Creating resources with label  ...
    error: ImageStream.image.openshift.io "wordpress" is invalid: metadata.labels: Invalid value: "qwer1345%3902#": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')
    error: DeploymentConfig.apps.openshift.io "wordpress" is invalid: [metadata.labels: Invalid value: "qwer1345%3902#": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]'), spec.template.labels: Invalid value: "qwer1345%3902#": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')]
    error: Service "wordpress" is invalid: [metadata.labels: Invalid value: "qwer1345%3902#": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]'), spec.selector: Invalid value: "qwer1345%3902#": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')]
```

Reply:
and yeah, our CI jobs that run against PRs run these tests, so your PR won't merge if they are not correct.